### ISSUE TITLE
Sort todo list

### ISSUE THREAD

**Issue by sadtab** [nvim-neorg/neorg#1434]
### Issues

- [X] I have checked [existing issues](https://github.com/vhyrro/neorg/issues?q=is%3Aissue) and there are no existing ones with the same request.

### Feature description

The title says it all, a feature that sorts the todo list, the most common sorting is to do bring "undone"s top and "done"s to the bottom  and others somewhere in between

### Help

No

### Implementation help

Shame on me, I don't know lua very well

**Comment by Anrock** [comment]
It's kinda already there?
```
- ( ) test
- (x) test
- (-) test
- (x) test
- ( ) test
- (_) test
- (=) test
```
`:sort` turns it into
```
- ( ) test
- ( ) test
- (-) test
- (=) test
- (_) test
- (x) test
- (x) test
```
Order is: not started, started, on pause, cancelled, done.

**Comment by sadtab** [comment]
@Anrock, not exactly, it works fine for homo-intended tasks, but it will completely mess with the subtasks, a proper sorting requires to implement the logic to to take this into account  
```
- ( ) task1
-- (x) sub1
-- (-) sub2
-- ( ) sub3
- (x) task2
- ( ) task3
- (_) task4
- (=) task5
- (x) task6
-- (x) sub7
-- (x) sub8
```
to 

```
```
- ( ) task1
-- ( ) sub3
-- (-) sub2
-- (x) sub1
- ( ) task3
- (_) task4
- (=) task5
- (x) task2
- (x) task6
-- (x) sub7
-- (x) sub8
```

**Comment by Anrock** [comment]
Ah, indeed. I totally forgot about subitems

**Comment by agusdmb** [comment]
i would love this too
